common/options.go

```diff
@@ -0,0 +1,58 @@
+package common
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/dop251/goja"
+)
+
+func parseBoolOpt(key string, val goja.Value) (b bool, err error) {
+	if b, err = strconv.ParseBool(val.String()); err != nil {
```

Comment:
That won't be an issue because `ParseBool` handles `"true"`, `1`, `true`, and others as well. Please see [ParseBool](https://pkg.go.dev/strconv#ParseBool) in the Go documentation.

Reply:
It does feel slightly safer to use the `.ToBoolean()` function that is exported from goja -- there's no guarantee as to how a bool is represented within goja, and maybe it's best to let it parse to bool.

Reply:
Goja's `.ToBoolean()` won't allow us to check whether the value is a `bool`. And it has problems for this case :-( For example, when you pass it any string such as "hello", it returns `true`. So I don't think it'd be a wise choice here.

Reply:
Ah, sounds like it's using JS boolean logic within `ToBoolean`.

Happy to leave it as is. I don't know what the pros and cons are of either 🤷

Reply:
Right.

It should be clear from the context, but let me try to reiterate.

`val.Export().(bool)` (after the type check of course):
* Pros: One way of accepting a bool value (`true` or `false` or raise an error). Consistency.
* Cons: Less flexibility for users. They can't just set `headless` (i.e.) to `"true"`. They need to set it to `true`.

`ParseBool`:
* Pros: [Many ways](https://pkg.go.dev/strconv#ParseBool) of accepting a bool value as long as the value is boolean-able (parsable by `ParseBool`). Flexibility for users.
* Cons: Less consistency overall and maybe more documentation work for us (?).

Reply:
I think this is should be a UX decision, not a technical one. I doubt that users would find the flexibility of passing `"true"` or `1` useful, and actual booleans would be safer, easier to understand and type check, etc.

This goes against parsing duration values, as we do want to allow duration strings or ms integers there, but I think that's a UX gain in that case.

In general, I think we should fallback to whatever behavior k6 currently has. From what I can see, they don't allow passing anything but a boolean for such fields. For example, setting `noConnectionReuse: 1` fails with `cannot unmarshal float64 into Go value of type null.Bool`. (Sidenote: we might also want to start using nullable types.)

Reply:
👍 So, from what I got from this discussion, we'll leave `parseBool` as it is (strict)?

> Sidenote: we might also want to start using nullable types.

I'll fix it in the following PR without those, but I'll check them out later; thanks!

Reply:
But it's currently _not_ strict... I think we should only accept actual boolean values.

Reply:
Resolved in 2f9a61a.